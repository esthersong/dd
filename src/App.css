
.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
html, body, #root, .App, .chat-container, .chatrooms, .set-username{
  height: 100%;
  width: 100%;
  font-weight: lighter;
}
input:focus,
form button:focus {
  outline: 0;
}
div.container {
  max-width: 100%;
  padding: 0;
}
.chat-container {
  width: 100%;
}
  .set-username {
    padding-top: 18%;
  }
  .set-username form {
    width: 350px;
    margin: 0 auto;
  }
    .set-username form input,
     .set-username form button{
      width: 100%;
      padding: 8px 10px;
    }
    .set-username form input {
      border: 1px solid #e5e5e5;
      font-size: 14px;
    }
    .set-username form button {
      margin-top: 20px;
      border: 0;
      background-color: #fc2046;
      color: #fff;
      font-weight: lighter;
      font-size: 14px;
    }
  .chatrooms {
    display: flex;
    width: 100%;
    position: relative;
  }
.chat-menu, .chatroom, .profile {
  display: inline-block;
  vertical-align: top;
}
.profile {
  text-align: center;
  padding-top: 5%;
}
  .profile img {
    max-height: 200px;
    opacity: 0.5;
  }
.chat-menu {
  height: 100%;
  flex: 0 220px;
  text-align: left;
  background-color: #fc2046;
  color: #fff;
}
  .user {
    height: 14%;
    padding: 20px;
    min-height: 80px;
  }
    .name {
      font-weight: bold;
    }
    .minutes {
      font-size: 12px;
    }
  .menu-item {
    padding: 15px 0px;
    padding-left: 20px;
    font-size: 14px;
  }
  .menu-item:hover {
    background-color: #db1839;
  }
  .selected,
  .selected:hover {
    background-color: #7e1022;
  }
.chatroom, .profile {
  flex: 1;
  height: 100%;
}
  .info {
    position: relative;
    height: 14%;
    min-height: 80px;
    padding: 20px 0px;
    text-align: center;
    -webkit-box-shadow: 0px 6px 4px -5px rgba(0,0,0,0.2);
    -moz-box-shadow: 0px 6px 4px -5px rgba(0,0,0,0.2);
    box-shadow: 0px 6px 4px -5px rgba(0,0,0,0.2);
    z-index: 10;
    color: #565656;
  }
    .info .chatroom-name {
      font-size: 18px;
      font-weight: normal;
    }
    .info .chatroom-members {
      font-size: 12px;
    }
  .message-board {
    padding: 5px 15px;
    background-color: #eff1f2;
    height: 74%;
    overflow: scroll;
  }
  .message-box {
    position: relative;
    width: 100%;
    overflow: hidden;
    font-size: 14px;
    margin-top: 5px;
  }
    .my-msg, .user-msg {
      max-width: 40%;
      word-wrap: break-word;
      margin-bottom: 10px;
    }
      .message {
        padding: 10px 20px;
        border-radius: 25px;
      }
      .my-msg {
        float: right;
      }
        .my-msg .message {
          margin-left: auto;
          background-color: #fc2046;
          color: #fff;
        }
        .my-msg .message-name {

        }
      .user-msg {
        float: left;
      }
        .user-msg .message {
          background-color: #fff;
          color: #000;
        }
    .message-name {
      margin: 5px 0px;
      color: #848484;
    }
  .chatroom form {
    height: 12%;
    min-height: 65px;
    padding: 0px 20px;
    padding-top: 2%;
    width: 100%;
    /* background-color: blue; */
  }
    .chatroom form input {
      width: 86%;
      height: 75%;
      padding-left: 10px;
      font-size: 12px;
    }
    .chatroom form button {
      width: 13%;
      border: 0px;
      border: none;
      background-color: #fff;
      margin-left: 5px;
      color: #4a91de;
    }
    .chatroom form button:focus {
      color: #316baa;
    }
