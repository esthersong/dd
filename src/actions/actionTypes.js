export const SET_USERNAME = 'SET_USERNAME';
export const SHOW_PROFILE = 'SHOW_PROFILE';
export const GET_ROOMS = 'GET_ROOMS';
export const GET_ROOM_DATA_STARTED = 'GET_ROOM_DATA_STARTED';
export const GET_ROOM_DATA_SUCCESS = 'GET_ROOM_DATA_SUCCESS';
export const POST_MESSAGE = 'POST_MESSAGE';
