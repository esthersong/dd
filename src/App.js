import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import ChatContainer from './components/ChatContainer';

class App extends Component {
  render() {
    return (
      <div className="App">
        <ChatContainer />
      </div>
    );
  }
}

export default App;
