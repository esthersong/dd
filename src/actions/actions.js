import * as types from './actionTypes';
import axios from 'axios';
import * as Constants from '../constants';

function getRoomsSuccess(data) {
  return {
    type: types.GET_ROOMS,
    rooms: data
  }
}

function getRoomDataStarted() {
  return {
    type: types.GET_ROOM_DATA_STARTED,
    loader: true,
    showChatroom: true
  }
}
function getRoomDataSuccess(data) {
  return {
    type: types.GET_ROOM_DATA_SUCCESS,
    loader: false,
    room: data.room,
    messages: data.messages
  }
}

function postMessageSuccess(data, id) {
  return {
    type: types.POST_MESSAGE,
    roomId: id,
    newMsg: data
  }
}

export const setUsername = (username) => {
    return {
      type: types.SET_USERNAME,
      username: username
    }
}

export const showProfile = () => {
  return {
    type: types.SHOW_PROFILE,
    showChatroom: false
  }
}
export const getRooms = () => {
  return function (dispatch) {
    return axios.get(`${Constants.API}rooms`)
    .then(data => {
      console.log('rooms list: ', data.data);
      dispatch(getRoomsSuccess(data.data));
    }).catch(err => {
      console.log('error: ', err);
    });
  }
}

export const getRoom = (id) => {
  return function (dispatch) {
    let promises = [];
    dispatch(getRoomDataStarted());
    promises.push(new Promise((resolve, reject) => {
      axios.get(`${Constants.API}rooms/${id}`)
      .then(data => {
        // console.log('success getting room data: ', data);
        resolve(data.data);
      }).catch(err => {
        console.log('error getting room: ', err);
        reject(err);
      })
    }));
    promises.push(new Promise((resolve, reject) => {
      axios.get(`${Constants.API}rooms/${id}/messages`)
      .then(data => {
        resolve(data.data);
      }).catch(err => {
        console.log('error: ', err);
        reject(err);
      })
    }))
    return Promise.all(promises).then(data => {
      let obj = {
        room: data[0],
        messages: data[1]
      };
      console.log('room and messages data: ', data);
      dispatch(getRoomDataSuccess(obj))
    }).catch(err => {
      console.log('err: ', err);
    })
  }
}

export const postMessage = (id, name, message) => {
  return function (dispatch) {
    return axios.post(`${Constants.API}rooms/${id}/messages`, {
      name: name,
      message: message
    })
    .then(data => {
      console.log('posted message successfully: ', data.data);
      dispatch(postMessageSuccess(data.data, id));
    }).catch(err => {
      console.log('error: ', err);
    });
  }
}
