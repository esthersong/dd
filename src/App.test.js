import React from 'react';
import ReactDOM from 'react-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';
import expect from 'expect';
import App from './App';
import * as actions from './actions/actions';
import * as types from './actions/actionTypes';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('actions', () => {
  it('should create an action to add a message', () => {
    const id = 0;
    const name = 'username';
    const message = "This is a test message";
    const store = mockStore({});
    const expectedMessage = message;
    const expectedName = name;
    const expectedAction = types.POST_MESSAGE;

    return store.dispatch(actions.postMessage(id, name, message)).then(() => {
      expect(store.getActions()[0]["newMsg"]["message"]).toEqual(expectedMessage);
      expect(store.getActions()[0]["newMsg"]["name"]).toEqual(expectedName);
      expect(store.getActions()[0]["type"]).toEqual(expectedAction);
    })
  });

  it('should create an action to get chatrooms', () => {
    const store = mockStore({});
    return store.dispatch(actions.getRooms())
    .then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it("should create an action to get a chatroom's info", () => {
    const id = 0;
    const store = mockStore({});
    return store.dispatch(actions.getRoom(id))
    .then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });


})
