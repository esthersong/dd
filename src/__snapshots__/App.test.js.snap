// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create an action to get a chatroom's info 1`] = `
Array [
  Object {
    "loader": true,
    "showChatroom": true,
    "type": "GET_ROOM_DATA_STARTED",
  },
  Object {
    "loader": false,
    "messages": Array [
      Object {
        "id": "gg35545",
        "message": "ayyyyy",
        "name": "Ryan",
        "reaction": null,
      },
      Object {
        "id": "yy35578",
        "message": "lmao",
        "name": "Nick",
        "reaction": null,
      },
      Object {
        "id": "hh9843",
        "message": "leggooooo",
        "name": "Danielle",
        "reaction": null,
      },
      Object {
        "id": "zGcyERazI",
        "message": "test",
        "name": "Ryan",
        "reaction": null,
      },
      Object {
        "id": "EZlu2JQ-Ey",
        "message": "tes",
        "name": "Ryan",
        "reaction": null,
      },
      Object {
        "id": "iz5WycOZV",
        "message": "tes",
        "name": "Ryan",
        "reaction": null,
      },
      Object {
        "id": "HAqm4sPG8o",
        "message": "tes",
        "name": "Ryan",
        "reaction": null,
      },
      Object {
        "id": "gD7VMM2we",
        "message": "tes",
        "name": "Ryan",
        "reaction": null,
      },
      Object {
        "id": "v-qptq15E",
        "message": "dsf",
        "name": "Esther Song",
        "reaction": null,
      },
      Object {
        "id": "WGoFBOyDg",
        "message": "Hey what's up?",
        "name": "esther song",
        "reaction": null,
      },
      Object {
        "id": "r8Pbzvt5l",
        "message": "hello",
        "name": "e",
        "reaction": null,
      },
      Object {
        "id": "UZIYDIbKF",
        "message": "hey",
        "name": "esther",
        "reaction": null,
      },
      Object {
        "id": "UR5rXUrR1",
        "message": "hey there",
        "name": "Albert",
        "reaction": null,
      },
      Object {
        "id": "7phNrHXV8",
        "message": "hey there",
        "name": "Albert",
        "reaction": null,
      },
      Object {
        "id": "n50AzhjrJ",
        "message": "hello",
        "name": "Albert",
        "reaction": null,
      },
      Object {
        "id": "7_gfojOu5",
        "message": "This is a test message",
        "name": "username",
        "reaction": null,
      },
    ],
    "room": Object {
      "id": 0,
      "name": "Tea Chats",
      "users": Array [
        "Ryan",
        "Nick",
        "Danielle",
        "Esther Song",
        "esther song",
        "e",
        "esther",
        "Albert",
        "username",
      ],
    },
    "type": "GET_ROOM_DATA_SUCCESS",
  },
]
`;

exports[`actions should create an action to get chatrooms 1`] = `
Array [
  Object {
    "rooms": Array [
      Object {
        "id": 0,
        "name": "Tea Chats",
      },
      Object {
        "id": 1,
        "name": "Coffee Chats",
      },
    ],
    "type": "GET_ROOMS",
  },
]
`;
